internal/actions/sync_branch.go

```diff
@@ -248,6 +257,16 @@ func syncBranchRebase(
 		// Note that we've introduced B into the history of stacked-2, but
 		// not C or D since those commits come after M.
 		newUpstreamCommitHash := origParentBranch.MergeCommit
+		var origUpstream string
+		if origParent.Trunk {
+			var err error
+			origUpstream, err = repo.RevParse(&git.RevParse{Rev: "origin/" + origParent.Name})
```

Comment:
for my understanding, if i read the `BranchState` correctly, the `BranchState.Head` represents the SHA of the parent branch, not the SHA of the current branch. But `git rev-parse` gives you SHA of the current branch.

Is that an intended change?

Reply:
BranchState.Head is empty if Trunk = true. Hence this change.